libcudacxx/include/cuda/__type_traits/is_floating_point.h

```diff
@@ -34,10 +34,9 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT is_floating_point
 {};
 
 #if !defined(_CCCL_NO_VARIABLE_TEMPLATES)
+// users are allowed to specialize is_floating_point, so define is_floating_point_v in terms of it
 template <class _Tp>
-_CCCL_INLINE_VAR constexpr bool is_floating_point_v =
-  _CUDA_VSTD::is_floating_point_v<_CUDA_VSTD::remove_cv_t<_Tp>>
-  || _CUDA_VSTD::__is_extended_floating_point_v<_CUDA_VSTD::remove_cv_t<_Tp>>;
+_CCCL_INLINE_VAR constexpr bool is_floating_point_v = is_floating_point<_Tp>::value;
```

Comment:
With C++17 being a requirement we should do it the other way around, we should define the struct through the variable template because we effectively always want to use that

Reply:
Can we make `cuda::is_floating_point` an alias to an internal struct namespace? This way, a user cannot even specialize it.